php/src/Google/Protobuf/Internal/DescriptorProto.php

```diff
@@ -71,9 +71,29 @@ class DescriptorProto extends \Google\Protobuf\Internal\Message
     private $reserved_name;
     private $has_reserved_name = false;
 
-    public function __construct() {
+    /**
+     * Constructor.
+     *
+     * @param array $data {
+     *     Optional. Data for populating the Message object.
+     *
+     *     @type string $name
```

Comment:
Do we need info for all fields in doc? This seems too much for me. User should be able to figure out the correct through field accessors.

Reply:
They can, but we want to help users as much as possible. Otherwise they will have to search the entire class looking at getters/setters to see what options are supported, and also they won't know necessarily to use `nested_type` vs. `nestedType`.

Reply:
I prefer a general guide here, e.g., use the original field name defined in proto file.
If field amount is small, then there is small overhead to look at the file. If field amount is large, then looking at doc here will be the same as looking at the whole file. In our experience, it's quite common that user define hundreds of fields in a single message. To that case, the doc will be unreasonably huge.

Reply:
Or shall we first check how many fields are there? and then decide whether to generate the doc?

Reply:
Besides, I think only field name and type is needed. User should go to the doc of accessor for detailed info.

Reply:
@TeBoring I think it is important to provide the constructor documentation accurately, even if that means a long list of fields. One thing we could do is, if the protobuf has say 100 fields, we could provide only one-line for it, and omit the description comment from the constructor, e.g.

```
## With < 100 fields 
@param int count
       The number of things that make up something. This is the description that exists
       in the proto file before the field, if specified.

## With > 100 fields 
@param int count
```

Reply:
ruby also supports array constructor. However, it doesn't provide any info about params. Besides, previously, we also don't provide detailed info at accessors. Can we postpone this to another PR?

Reply:
For java, it's the same.

Reply:
I think Java uses the builder pattern, so all of the constructors are private.
I am not enough of a Rubyist to interpret what is going on in this file - it looks like the protobuf classes are dynamic, and have no documentation whatsoever? https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-video_intelligence/lib/google/cloud/videointelligence/v1/video_intelligence_pb.rb
Python seems to generate something in its docs that lists the supported "attributes": https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/videointelligence/google/cloud/videointelligence_v1/proto/video_intelligence_pb2.py#L1132
Go uses structs, e.g.: https://github.com/google/go-genproto/blob/master/googleapis/cloud/videointelligence/v1/video_intelligence.pb.go#L169

To be honest I'm not really sure what the right comparison language is. Both python and ruby seem to support "dynamic" protobuf classes in some sense; Java uses builders; Go uses structs.

Reply:
1) My feeling to this change is that is relatively trivial. And it's also not clear to how many users will think it useful and how many users will think it bothering.
2) This is not a blocking change. We can also add it later without affecting any existing users.
3) In all other languages, protobuf is also exposed as API for other libraries. We haven't been asked for that feature yet. From this experience, I don't think it's very important to add it.

We can add it later if we find it is very important. Right now, if you want your library's proto generated file have that, could you add it in your own exposed file?

Reply:
I am ok with adding these docs. Users are fully responsible for the doc content (e.g., if docs are too long, we are not responsible to make it shorter. Users need to control their proto files.)